test/controllers/EvidenceDownloadController.test.ts

```diff
@@ -0,0 +1,107 @@
+import 'reflect-metadata';
+
+import Substitute from '@fluffy-spoon/substitute';
+import { SessionStore } from 'ch-node-session-handler';
+import { expect } from 'chai';
+import { GATEWAY_TIMEOUT, INTERNAL_SERVER_ERROR, NOT_FOUND, OK } from 'http-status-codes';
+import { Readable } from 'stream';
+import request from 'supertest';
+
+import 'app/controllers/EvidenceDownloadController';
+import { FileMetadata } from 'app/models/FileMetadata';
+import { FileTransferService } from 'app/service/FileTransferService';
+import { DOWNLOAD_FILE_PAGE_URI } from 'app/utils/Paths';
+
+import { createAppConfigurable } from 'test/ApplicationFactory';
+const createDefaultApp = (fileTransferService: FileTransferService) => createAppConfigurable(container => {
+    container.bind(SessionStore).toConstantValue(Substitute.for<SessionStore>());
+    container.bind(FileTransferService).toConstantValue(fileTransferService);
+});
+describe('EvidenceDownloadController', () => {
+    const FILE_ID = '123';
+    const DOWNLOAD_PROMPT_URL = `${DOWNLOAD_FILE_PAGE_URI}/prompt/${FILE_ID}`;
+    const EXPECTED_DOWNLOAD_LINK_URL = `${DOWNLOAD_FILE_PAGE_URI}/data/${FILE_ID}/download`;
+
+    const contentDisposition = `attachment; filename=hello.txt`;
+    const contentLength = 1000;
+    const contentType = `application/json`;
+
+    const metadata: FileMetadata = {
+        av_status: 'scanned',
+        content_type: contentType,
+        id: FILE_ID,
+        name: 'hello.txt',
+        size: contentLength
+    };
+
+    const fileTransferServiceProxy = (downloadResult: Promise<Readable>): FileTransferService => {
```

Comment:
Out of curiosity, why didn't you use SubstituteJs?

Reply:
At one point in time I had to actually use Proxy to do something custom like adding a custom callback when I had one. But now, yes I could just use SubstituteJs

Reply:
I suggest we use SubstituteJS fo consistency to not set an exception that could be copied over and over and spread like virus :)